desktop/install/linux-install.md

```diff
@@ -268,6 +270,19 @@ Docker Desktop for Linux runs a Virtual Machine (VM) for the following reasons:
 
     As such, we have adjusted the default memory available to the VM in DD4L. You can tweak this setting to your specific needs by using the **Memory** slider within the **Settings** > **Resources** tab of Docker Desktop.
 
+## File sharing
+
+Docker Desktop for Linux uses [virtiofs](https://virtio-fs.gitlab.io/) as the default (and currently only) mechanism to enable file sharing between the host and Docker Desktop VM. In order not to require elevated privileges, without unnecessarily restricting operations on the shared files, Docker Desktop runs the file sharing service (`virtiofsd`) inside a [user namespace](https://man7.org/linux/man-pages/man7/user_namespaces.7.html) with UID and GID mapping configured. As a result Docker Desktop will rely on the host to be configured to enable the current user to use subordinate ID delegation. For this to be true [/etc/subuid](https://man7.org/linux/man-pages/man5/subuid.5.html) and [/etc/subgid](https://man7.org/linux/man-pages/man5/subgid.5.html) have to be present (Docker Desktop currently only supports subordinate ID delegation configured via files). In case of both Docker Desktop will use the first entry corresponding to the current user, to set up mappings for IDs above 0, and map the current users ID and GID to 0.
+
+In case `/etc/subuid` and `/etc/subgid` are missing they will need to be created. Both should contain entries in the following form `<username>:<start of id range>:<id range size>`. E.g. to allow a user whose username is `myusername` to use IDs from 100000 to 165535:
+
+```console
+$ echo 'myusername:100000:65536' | sudo tee /etc/subuid
+$ echo 'myusernamer:100000:65536' | sudo tee /etc/subgid
+```
+
+In this scenario if a shared file is `chown`ed inside a Docker Desktop container to be owned by a user with a UID of 1000 it will show up on the host as owned by a user with a UID of 100999. This has the unfortunate side effect of preventing easy access to such a file on the host. The problem can be resolved by creating a group with the new GID and adding our user to it, or by setting a recursive [ACL](https://man7.org/linux/man-pages/man1/setfacl.1.html) on folders shared with the Docker Desktop VM.
```

Comment:
IMO this is unexpected enough (from a user perspective) that we should _really_ reconsider whether this is the default behavior we want and/or whether this is the _only_ behavior we want to support -- the two obvious alternatives I see (either of which I personally would opt for) are running just `virtiofs` privileged (small security boundary, by design) or running it entirely unprivileged so that all shared files are "stuck" being owned by the host user which I think matches the behavior on Windows and macOS better, right?

(Sorry if this conversation already happened / is already happening elsewhere! :see_no_evil:)

As a side note (and probably the reason @thaJeztah pulled me into a thread here), I chuckled a little at "In order not to require elevated privileges," because setting this up is exactly the thing that _did_ require elevated privileges for me. :sweat_smile:

Reply:
> As a side note (and probably the reason @thaJeztah pulled me into a thread here), I chuckled a little at "In order not to require elevated privileges," because setting this up is exactly the thing that _did_ require elevated privileges for me. 😅

Yeah this is unfortunate :( On most distros this will be the default configuration now, so in a good number of cases users shouldn't need to do anything to configure subordinate IDs.

> IMO this is unexpected enough (from a user perspective) that we should _really_ reconsider whether this is the default behavior we want and/or whether this is the _only_ behavior we want to support -- the two obvious alternatives I see (either of which I personally would opt for) are running just `virtiofs` privileged (small security boundary, by design) or running it entirely unprivileged so that all shared files are "stuck" being owned by the host user which I think matches the behavior on Windows and macOS better, right?

I do agree that introducing a behaviour different to both, the native engine and DD on other OSes, should be avoided.

Option 1 would require asking the user for the `root` password every time we start (or they add a share), which is not great UX-wise (and goes against the requirement to run `root`-less), and has potential to adversely affect the host environment.

As for option 2 - it would be optimal, but is still a couple releases away from being possible. @fredericdalleau is working on a "fake owner" driver to use with virtiofs on macOS, where currently changing shared file ownership from the VM does not work, and changing ownership on the host file system, in the way we do on Linux, isn't an option. Once that's done, my intention is to make it the default behaviour on Linux.

As a middle ground I am considering having DD assign ACLs to shared folders at the moment of bind mounting. It would decrease the inconvenience of dealing with the actual rights for some files being changed, but introduces yet another assumption about the host environment.